src/cmd/script.rs

```diff
@@ -864,6 +871,13 @@ fn emit_vivado_tcl(
             .iter()
             .map(|t| (format!("TARGET_{}", t.to_uppercase()), None)),
     );
+    if let Some(define) = matches.values_of("defines") {
+        defines.extend(
+            matches.values_of("defines")
+            .unwrap()
+            .map(|t| (t.to_string(), None))
```

Comment:
This does not support assigning a value to the define at the moment. I think adding this functionality should be fairly straightforward:
```suggestion
            .map(|t| {
                let mut parts = t.splitn(2, "=");
                let name = parts.next().unwrap().trim(); // split always has at least one element
                let value = parts.next().map(|v| v.trim());
                (name.to_string(), value.to_string())
            })
```
The `splitn` gives you an iterator over the parts of the string separated by "=", and at most 2 (with the second entry containing the remainder of the string). The `trim` then cuts away any potential leading and trailing whitespaces.

Reply:
It allowed assigning a value as the first entry of the tuple was already the whole string. That is, with `--define A=0`, the tuple became `("A=0", None)`. But I agree this is cleaner, especially for targets that don't use a equal sign for their assignment (if those targets exist). I also allows for stuff like this: `--define "A = 0"`.

Or did you mean something different with supporting to assign values?

Reply:
Exactly what I meant :+1: